### ISSUE TITLE
% grid entry

### ISSUE THREAD

**Issue by ethicryd** [chrisleekr/binance-trading-bot#517]
I am wondering if I have understood right that the % entry of grid levels is calculated from "Last Buy Price", and if I am right why is it done like this and not from previous grid entry price?

**Comment by chrisleekr** [comment]
@ethicryd 

The first grid's trigger price will be calculated based on the lowest price.
From the second grid, it will calculate based on the last buy price.

![image](https://user-images.githubusercontent.com/5715919/194534520-fd825f6f-8280-4bed-a53b-8b27d2059fb8.png)

Please close the issue if I answered your question.

**Comment by ethicryd** [comment]
Yes, I could see this buy what I was wondering if the "Last Buy Price" is actually what you call "Last Buy Price" when you refer to the average buy price that is used to calculate the exit point.
Is this the case?

Or is the "last buy price", in the case of the grid entry, the actual last price at which the bot bought last grid level?

**Comment by ethicryd** [comment]
I am not sure how to explain this, but if you use the Last Buy Price (intended as current the average entry price), you may end u in a situation where several of your grid levels gets trigger all at once. Hope what I am saying makes sense and if necessary I can explain further.

On the other end if you base the trigger of the next grid on the entry price of the previous grid, then you can have control.

**Comment by peweb** [comment]
> I see that the "last buy price" is actually the recalculated average entry price. Why was made this choice to use the average price for the entry in the grid level, and not last price at which the previous grid level was bought?

Yes, you are right. I do believe we should have an option to choose between last buy price or last grid price. 

![image](https://user-images.githubusercontent.com/36623351/195825732-85c49940-4b5f-4a76-b140-c8c7850505e0.png)

**Comment by ethicryd** [comment]
Thank you.
Would you manage to implement the option?

**Comment by chrisleekr** [comment]
@ethicryd Sorry, I don't have time to add a new feature. 
If you or any developers can contribute, I welcome it. :)

**Comment by chrisleekr** [comment]
Closing this as a refactored v1 has been released. It's no longer relevant to the latest code.
